### ISSUE TITLE
Support using different font family variants depending on user language

### ISSUE THREAD

**Issue by Robyt3** [ddnet/ddnet#6881]
https://heistak.github.io/your-code-displays-japanese-wrong/

TL;DR: There are characters (e.g. all in `刃直海角骨入`) that look different depending on the language of the content being Japanese, Simplified Chinese, Traditional Chinese and Hangul, because Unicode uses the same codepoint for characters regardless of the language.

This is how it looks with DDNet client currently on all languages:

- ![5](https://github.com/ddnet/ddnet/assets/23437060/b8ebe4b8-98e9-446f-ae1d-65e50a100704)

Upstream does this correctly already thanks to @TsFreddie (teeworlds/teeworlds#2704), so this is how it looks there:

- Japanese:
![1](https://github.com/ddnet/ddnet/assets/23437060/cfe59c47-c49b-408b-8b37-f5e6298a60da)
- Simplified Chinese:
![2](https://github.com/ddnet/ddnet/assets/23437060/5405e129-dc32-49dc-b4ee-2a6f351c2c25)
- Traditional Chinese:
![3](https://github.com/ddnet/ddnet/assets/23437060/21e2f278-3742-4315-aac1-972bfe8ff57d)
- Hangul:
![4](https://github.com/ddnet/ddnet/assets/23437060/2a6b9669-ff05-49a9-a0ec-4212040f6df7)

The idea is to port the "Add multi-font fallback support as well as regional variant switching" changes from upstream including the font index file.

The index is a json file looking like this:

```json
{
    "font files": [
        "DejaVuSans.ttf",
        "SourceHanSans.ttc"
    ],
    "default": "DejaVu Sans",
    "language variants": {
        "japanese": "Source Han Sans",
        "korean": "Source Han Sans K",
        "simplified_chinese": "Source Han Sans SC", 
        "traditional_chinese": "Source Han Sans TC"
    },
    "fallbacks": [
        "Source Han Sans"
    ]
}
```

We would also add another entry for the icon font.

Looks like upstream uses Source Han Sans for all CJK languages, whereas we currently use Glow Sans Compressed for Japanese and Source Han Sans for Simplified Chinese.

**Comment by heinrich5991** [comment]
That's cool. :)

**Comment by TsFreddie** [comment]
FYI, Glow Sans Compressed is chose to better fit long japanese phrases into the buttons.
